src/main/java/com/internet/cinema/dao/implementation/UserDaoImpl.java

```diff
@@ -0,0 +1,51 @@
+package com.internet.cinema.dao.implementation;
+
+import com.internet.cinema.dao.UserDao;
+import com.internet.cinema.exception.DataProcessingException;
+import com.internet.cinema.lib.Dao;
+import com.internet.cinema.model.User;
+import com.internet.cinema.util.HibernateUtil;
+import java.util.Optional;
+import org.apache.log4j.Logger;
+import org.hibernate.Session;
+import org.hibernate.Transaction;
+import org.hibernate.query.Query;
+
+@Dao
+public class UserDaoImpl implements UserDao {
+    private static final Logger LOGGER = Logger.getLogger(UserDaoImpl.class);
+
+    @Override
+    public User add(User user) {
+        Session session = HibernateUtil.getSessionFactory().openSession();
+        Transaction transaction = null;
+        try {
+            transaction = session.beginTransaction();
+            Long userId = (Long) session.save(user);
+            user.setId(userId);
+            transaction.commit();
+            LOGGER.info("User with id " + userId + " was added.");
+            return user;
+        } catch (Exception e) {
+            if (transaction != null) {
+                transaction.rollback();
+            }
+            throw new DataProcessingException("Can`t add user.", e);
+        } finally {
+            if (session != null) {
+                session.close();
+            }
+        }
+    }
+
+    @Override
+    public Optional<User> findByEmail(String email) {
+        try (Session session = HibernateUtil.getSessionFactory().openSession()) {
+            Query<User> query = session.createQuery("FROM User WHERE email = :eml");
+            query.setParameter("eml", email);
+            return query.getResultStream().findFirst();
```

Comment:
```suggestion
            return query.uniqueResult();
```
Please, read the group chat in Slack